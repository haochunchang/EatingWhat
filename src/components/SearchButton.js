import React, { Fragment } from "react";

const SearchButton = ({ id, isDecided, onClicked }) => {
  return isDecided ? (
    <Fragment>
      <button
        id={id}
        type="button"
        className="f3 pa3 btn btn-success grow"
        onClick={onClicked}
      >
        Search Nearby Foods
      </button>
    </Fragment>
  ) : (
    <h2>Not yet decide what to eat? Click buttons to decide!</h2>
  );
};

export default SearchButton;
