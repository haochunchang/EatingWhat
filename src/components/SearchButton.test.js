import React from "react";
import { shallow } from "enzyme";
import SearchButton from "./SearchButton";

describe("SearchButton", () => {
  it("matches not decided SearchButton snapshot", () => {
    expect(shallow(<SearchButton isDecided={false} />)).toMatchSnapshot();
  });

  it("matches decided SearchButton snapshot", () => {
    expect(shallow(<SearchButton isDecided={true} />)).toMatchSnapshot();
  });
});
