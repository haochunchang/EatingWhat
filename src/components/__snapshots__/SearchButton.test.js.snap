// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchButton matches decided SearchButton snapshot 1`] = `ShallowWrapper {}`;

exports[`SearchButton matches not decided SearchButton snapshot 1`] = `ShallowWrapper {}`;
